packages/E2ETest/windows/TreeDumpLibrary/VisualTreeDumper.cs

```diff
@@ -0,0 +1,179 @@
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using System.Text;
+using System.Threading.Tasks;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Media;
+
+namespace TreeDumpLibrary
+{
+    public sealed  class VisualTreeDumper
+    {
+        class Visitor
+        {
+            private DefaultVisualTreeLogger _logger;
+            private int _indent;
+            private DefaultFilter _filter;
+            private DefaultPropertyValueTranslator _translator;
+            public Visitor(DefaultFilter filter, DefaultPropertyValueTranslator translator, DefaultVisualTreeLogger logger)
+            {
+                _indent = 0;
+                _filter = filter;
+                _translator = translator;
+                _logger = logger;
+            }
+            public void EndVisitNode(DependencyObject obj)
+            {
+                _indent--;
+                _logger.EndNode(_indent, obj.GetType().FullName, obj);
+            }
+
+            public void BeginVisitNode(DependencyObject obj)
+            {
+                _logger.BeginNode(_indent, obj.GetType().FullName, obj);
+                _indent++;
+            }
+
+            public override string ToString()
+            {
+                return _logger.ToString();
+            }
+
+            public bool ShouldVisitPropertiesForNode(DependencyObject node)
+            {
+                return (node as UIElement) != null;
+            }
+
+            public bool ShouldVisitProperty(PropertyInfo propertyInfo)
+            {
+                return _filter.ShouldVisitProperty(propertyInfo.Name);
+            }
+
+            public void VisitProperty(string propertyName, object value)
+            {
+                var v = _translator.PropertyValueToString(propertyName, value);
+                if (_filter.ShouldVisitPropertyValue(v))
+                {
+                    _logger.LogProperty(_indent + 1, propertyName, v);
+                }
+            }
+        }
+
+        public static string DumpToXML(DependencyObject root, DependencyObject excludedNode)
```

Comment:
>DumpToXML [](start = 29, length = 9)

Please rename this. In the first design, the output is XML, but finally it's just a simple text with indent